components/ScrollingAvatars/ScrollingAvatars.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, useEffect, useState } from 'react';
+import Avatar from '@components/ScrollingAvatars/Avatar';
+import type { AvatarInfo } from './ScrollingAvatarsContainer';
+
+interface IProps {
+  /** pre-set width and height of each avatar */
+  size?: 'small' | 'mid' | 'large';
+  /** list of objects containing the configs for the avatars, maximum size is 24 */
+  avatarInfos: AvatarInfo[];
+  scrollDirection?: 'left' | 'right';
+  onMouseEnter?: () => void;
+  onMouseLeave?: () => void;
+  setName: (name: string) => void;
+  setPosition: (position: string) => void;
+}
+
+/** enum used to map size label and it's corresponding css properties.
+ * This enum must match that in ScrollingAvatar.scss */
+const AVATAR_ENUM = {
+  small: {
+    size: 60,
+    margin: 15,
+  },
+  mid: {
+    size: 100,
+    margin: 30,
+  },
+  large: {
+    size: 150,
+    margin: 50,
+  },
+};
+
+/**
+ * Given a list of avatar infos, calculate the total number of avatars needed to be appended to
+ * create infinte scrolling animation and return the new list of avatar infos.
+ */
+const appendAvatarInfos = (
+  avatarInfos: AvatarInfo[],
+  { size, margin }: { size: number; margin: number },
+) => {
+  const currAvatarWidth = avatarInfos.length * (size + 2 * margin);
+  const desiredWidth = window.innerWidth + 20 * currAvatarWidth;
+  const remainingWidth = desiredWidth - currAvatarWidth;
+  const remainingAvatars = Math.ceil(remainingWidth / currAvatarWidth);
+  const appendedAvatarsInfos = [...avatarInfos];
+
+  for (let i = 0; i < remainingAvatars; i += 1) {
+    appendedAvatarsInfos.push(avatarInfos[i % avatarInfos.length]);
+  }
```

Comment:
if it's not a bother, this could be rolled into a `forEach`

just to stay consistent with ES6 everywhere possible